.github/workflows/ci-post-land.yml

```diff
@@ -329,31 +329,31 @@ jobs:
 
   rustdoc:
     needs: prepare
-    if: ${{ needs.prepare.output.changes-target-branch == 'main' }}
-    env:
-      CARGO_INCREMENTAL: 0
-      RUSTFLAGS: -D warnings
+    if: ${{ needs.prepare.outputs.changes-target-branch == 'main' && needs.prepare.outputs.rust-changes == 'true'}}
     runs-on: ubuntu-latest
-
+    container:
+      image: diem/build_environment:${{ needs.prepare.outputs.changes-target-branch }}
+      volumes:
+        - "${{github.workspace}}:/opt/git/diem"
     steps:
       - name: Checkout repository
         uses: actions/checkout@v2.3.4
-
-      - name: Install Rust toolchain
-        uses: actions-rs/toolchain@v1
+      - uses: ./.github/actions/build-setup
+      - uses: actions/cache@v2.1.6
         with:
-          toolchain: stable
-          override: true
-
-      # Build the rust crate docs
-      # Use `RUSTC_BOOTSTRAP` in order to use the `--enable-index-page` flag of rustdoc
-      # This is needed in order to generate a landing page `index.html` for workspaces
+          path: "/opt/cargo/git\n/opt/cargo/registry\n/opt/cargo/.package-cache"
+          key: crates-${{ runner.os }}-${{ hashFiles('Cargo.lock') }}
+          restore-keys: "crates-${{ runner.os }}"
```

Comment:
Make consistent with other rust job's runtime (use the docker image and prepare, and cache step).